### PR TITLE
Python support update

### LINKED ISSUE #61: Update supported Python versions
- drop support for Python versions < 3.6
- add support for Python versions 3.7, 3.8 (if missing)
- fix linting errors (if present)

If you are a beginner, and need to help for this issue, especially during Hacktoberfest, please comment here or reach out to me via https://jugmac00.github.io/

### PR DESCRIPTION
**Pull request by sgaist**
This merge request updates the minimal Python version to 3.6 and adds 3.8.

It also adds the tools required to run the tests to list of dependencies as well fixes black and flake8 issues.

Fixes #61

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -2,9 +2,6 @@ dist: xenial
 language: python
 
 python:
-    - "3.4"
-    - "3.5"
-    - "3.6"
     - "3.7"
     - "3.8"
     - "pypy3.6-7.1.1"
```
Comment (jugmac00) [inline .travis.yml:7]: Could you please add back Python 3.6?

The newest version for pypy3  is.... pypy3 :-)

See https://docs.travis-ci.com/user/languages/python/#pypy-support

File: setup.py [hunk 2]
```diff
@@ -24,9 +24,6 @@
         "Intended Audience :: Developers",
         "License :: OSI Approved :: BSD License",
         "Topic :: Software Development :: Libraries :: Python Modules",
-        "Programming Language :: Python :: 3.4",
-        "Programming Language :: Python :: 3.5",
-        "Programming Language :: Python :: 3.6",
         "Programming Language :: Python :: 3.7",
```
Comment (jugmac00) [inline setup.py:27]: Please restore support for Python 3.6

File: tox.ini [hunk 3]
```diff
@@ -1,26 +1,26 @@
 [tox]
-envlist = py34, py35, py36, py37, py38, pypy3, coverage, pep8, docs, perf
+envlist = py37, py38, pypy3, coverage, pep8, docs, perf
 skipsdist = True
```
Comment (jugmac00) [inline tox.ini:3]: please restore support for python 3.6

File: tox.ini [hunk 4]
```diff
@@ -1,26 +1,26 @@
...
 skip_missing_interpreters = True
 
 [testenv]
 usedevelop = True
 extras = test
 
-commands = py.test {posargs}
+commands = pytest {posargs}
```
Comment (jugmac00) [inline tox.ini:11]: good catch!

**Review by jugmac00 — CHANGES_REQUESTED** [review]
Thank you very much for the PR.

LGTM with a few comments - please try to fix these.

File: .travis.yml (hunk 1 above)
Comment (sgaist) [reply .travis.yml:7]: Sure thing !

File: tox.ini (hunk 4 above)
Comment (sgaist) [reply tox.ini:11]: Thanks :)

File: tox.ini (hunk 3 above)
Comment (sgaist) [reply tox.ini:3]: Done

File: setup.py (hunk 2 above)
Comment (sgaist) [reply setup.py:28]: Done

**Comment by jugmac00** [comment]
Please, could you also add a change log entry?

**Comment by jugmac00** [comment]
It looks like you resolved most of the issues. One last thing - could you please add a change log entry? Thank you!

**Review by jugmac00 — APPROVED** [review]

**Comment by jugmac00** [comment]
LGTM! Thank you very much for your contribution and your patience! PR looks great.

**Pull request merged**